### PR TITLE
fix: correctly detect suffix for scoped packages with dots

### PR DESCRIPTION
**Pull request by rasimx**
Fixes an issue where `VirtualModule` failed to detect the correct suffix for scoped packages that contain dots in the namespace (e.g., `@company.x.y/lib.name`).

This caused the suffix to be incorrectly split from the namespace, resulting in broken imports.

Fixes: https://github.com/module-federation/vite/issues/321

### CODE REVIEW

File: src/utils/VirtualModule.ts [hunk 1]
```diff
@@ -30,6 +30,22 @@ function getNodeModulesDir() {
   return cachedNodeModulesDir;
 }
 
+function getSuffix(name: string): string {
```
Comment (gioboa) [inline src/utils/VirtualModule.ts]: It looks great to me, can you cover this function with a test with few different test cases?

Comment (rasimx) [reply src/utils/VirtualModule.ts]: Added unit tests for the getSuffix function to verify behavior with scoped packages and nested paths (e.g., .bin/@scope/pkg).

While writing tests, identified additional edge cases where the suffix was incorrectly inferred. Updated the implementation to handle these cases properly.

**Review by gioboa — APPROVED** [review]
Thanks, that's great @rasimx 
Kudos for your fist PR in this project :partying_face:
